client/connector_create.go

```diff
@@ -275,6 +213,33 @@ func (cli *VanClient) VerifySiteCompatibility(siteVersion string) error {
 	return nil
 }
 
+func (cli *VanClient) verifyNotSelfOrDuplicate(secret corev1.Secret, self string, options types.ConnectorCreateOptions) error {
+	if secret.ObjectMeta.Annotations == nil {
+		return fmt.Errorf("The secret has not annotations")
+	}
+	generatedBy, ok := secret.ObjectMeta.Annotations[types.TokenGeneratedBy]
+	if !ok {
+		return fmt.Errorf("Can't find secret origin.")
+	}
+	if self == string(generatedBy) {
+		return fmt.Errorf("Can't create connection to self with token")
+	}
+	currentSecrets, err := cli.KubeClient.CoreV1().Secrets(options.SkupperNamespace).List(metav1.ListOptions{LabelSelector: "skupper.io/type=connection-token"})
+	if err != nil {
+		return fmt.Errorf("Could not retrieve secrets: %w", err)
+	}
+	for _, currentSecret := range currentSecrets.Items {
+		currentAuthor, ok := currentSecret.Annotations[types.TokenGeneratedBy]
+		if !ok {
+			return fmt.Errorf("A secret has no author.")
```

Comment:
I see Skupper was already doing this earlier, but should it really return an error or
simply log an event and continue to the next iteration?

Reply:
That's a good question. If a secret has the connection-token label but does not have the annotation, it is exceptional based on what we currently expect (e.g. it "shouldn't" happen).

Reply:
True, it is an internal error or maybe some incorrect manual modification.

My concern was that new links would not be created, unless this is fixed,
but I believe it is better to fail fast in this case otherwise the event would
not be noticed.